### PR TITLE
Add more tf ops

### PR DESCRIPTION
**Pull request by alha02**
- reshape
- tile
- fill
- cast
- solve
- triangular_solve
- randu
- randn
- expand_dims

### CODE REVIEW

**Comment by MikeInnes** [comment]
Thanks!

File: src/ops.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export reshape, tile, fill, cast, solve, triangular_solve, randu, randn,
+       expand_dims
+
+import Base: reshape, fill, randn
+
+reshape(x::AbstractArray, dims::AbstractArray) = reshape(x,tuple(dims...))
```
Comment (tkelman) [inline src/ops.jl]: can this use a package-local name? it's encouraged not to extend Base functions on Base types, as that has global side effects and can change the behavior of unrelated code

Comment (tkelman) [inline src/ops.jl]: oh you got rid of this in https://github.com/MikeInnes/Flux.jl/commit/bfe1f8eeeece5349cdfcb632b2414d187dc7bb79 but that isn't included in your latest tag?

Comment (MikeInnes) [inline src/ops.jl]: Yeah, this isn't ideal but it's just needed as a stopgap. I'm doing a big refactor that'll get rid of the patching in the next version.

Comment (tkelman) [inline src/ops.jl]: k fair enough
